src/routes/preHandlers/activityHistories.js

```diff
@@ -8,6 +8,26 @@ const Course = require('../../models/Course');
 const Card = require('../../models/Card');
 const { SURVEY, OPEN_QUESTION, QUESTION_ANSWER } = require('../../helpers/constants');
 
+exports.checkQuestionnaireAnswersList = async (questionnaireAnswersList, activityId) => {
+  for (const qa of questionnaireAnswersList) {
+    const card = await Card.findOne({ _id: qa.card }).lean();
```

Comment:
plutot que de faire un appel par carte, tu pourrais faire un 
`const cards = Card.find({ _id: { $in: questionnaireAnswersList.map(qa => qa..card) } }).lean();`
puis tu boucles sur `cards`

Reply:
done